File: udemy/ds-and-a/DSAndA.Test/LinkedLists/Assignment4/RemoveDuplicates.Test.cs
using DSAndA.LinkedLists;
using Xunit;

namespace DSAndA.Test.LinkedLists.Assignment4
{
    public class RemoveDuplicates
    {
        [Fact]
        public void RemoveDuplicatesTest1()
        {
            LinkedList<int> linkedList = new LinkedList<int>();
            linkedList.InsertAtHead(1);
            linkedList.InsertAtHead(1);

            linkedList.RemoveDuplicates();

            Assert.Equal(expected: "1,", actual: linkedList.ToString());
        }

        [Fact]
        public void RemoveDuplicatesTest2()
        {
            LinkedList<int> linkedList = new LinkedList<int>();
            linkedList.InsertAtHead(1);
            linkedList.InsertAtHead(4);
            linkedList.InsertAtHead(3);
            linkedList.InsertAtHead(1);
            linkedList.InsertAtHead(3);
            linkedList.InsertAtHead(1);

            linkedList.RemoveDuplicates();

            Assert.Equal(expected: "1,3,4,", actual: linkedList.ToString());
        }
    }
}

File: udemy/ds-and-a/javascript-solutions/tests/LinkedLists/Assignment4/RemoveDuplicates.test.js
const LinkedList = require("../../../src/LinkedLists/LinkedList");

test('Remove duplicates from unsorted list', () => {
    let linkedList = new LinkedList();
    linkedList.insertAtHead(1);
    linkedList.insertAtHead(3);
    linkedList.insertAtHead(0);
    linkedList.insertAtHead(3);

    expect(linkedList.toString()).toBe('3, 0, 3, 1');

    linkedList.removeDuplicates();

    expect(linkedList.toString()).toBe('3, 0, 1');
});